### DISCUSSION TITLE
Transforming mouse cursor position to GeoPoint

### DISCUSSION THREAD

**Discussion by ComradeMashkov** [pelicanmapping/rocky#63] [Q&A / Support] [answered]
Greetings, Rocky developers and users. I was wondering: is it possible to convert mouse position coordinates, for example, received with ImGui (`ImGui::GetIO().MousePos`) to GeoPoint to extract latitude and longitude from it? I had several attempts but they did not led to any possible working solution. Is it possible by any chance? Or Rocky library does not have this functionality at the moment? Thanks!

**Comment by gwaldron** [comment, marked as answer]
You are in luck, I have just pushed two new functions, in the DisplayManager header.
```c++
GeoPoint pointAtWindowCoords(vsg::ref_ptr<vsg::View> view, int x, int y);
DisplayGeoPoint pointAtWindowCoords(vsg::ref_ptr<vsg::Viewer> viewer, int x, int y);
```
They both take window coordinates, which is what would typically be returned by ImGui or by a vsg PointerEvent.

(There is one catch; it doesn't seem to work for inset views yet. I am not sure why but I'll file a defect for it.)
